redundant-bmc/src/passive_role_handler.cpp

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: Apache-2.0
+#include "passive_role_handler.hpp"
+
+#include <phosphor-logging/lg2.hpp>
+
+namespace rbmc
+{
+
+constexpr auto bmcPassiveTarget = "obmc-bmc-passive.target";
+
+// NOLINTNEXTLINE
+sdbusplus::async::task<> PassiveRoleHandler::start()
+{
+    try
+    {
+        // NOLINTNEXTLINE
+        co_await services->startUnit(bmcPassiveTarget);
+    }
+    catch (const sdbusplus::exception_t& e)
+    {
+        // TODO: error log
```

Comment:
I can do that.

If it crashes and restarts, that'll be a natural retry, but if it the target never starts, then this service will die off and the BMC will be quiesced with no role property on D-Bus.  Hopefully the HMC can still see the BMC state and do the recovery like we had talked about.

Reply:
> this service will die off and the BMC will be quiesced

Have you planned to include the RBMC and CFAM manager service [here](https://github.ibm.com/openbmc/openbmc/blob/1110-ghe/meta-ibm/recipes-phosphor/state/phosphor-state-manager/phosphor-service-monitor-default.json#L2)?

> with no role property on D-Bus.
>
>  the HMC can still see the BMC state and do the recovery

Which recovery mechanism are you referring to?

I would like to ensure that no action is taken on the Active BMC if it's in a good state. The HMC needs a way to distinguish between a failover when the A-BMC is quiesced versus the P-BMC, using the Role property. Additionally, if the Role property is unavailable, as in this case, the P-BMC should be recovered.

Reply:
> Have you planned to include the RBMC and CFAM manager service here?

Yea, though I was thinking I'd wait until we make a new branch for pfuture. (I should probably make a story).

> Which recovery mechanism are you referring to?

In the workshop we discussed that the HMC would be able to reboot a Quiesced BMC.  It would have the BMC states of both BMCs, and the roles of both BMCs if this service is running, so it could have the logic of when to do a reboot vs a failover.

> and until recover the P-BMC, who will disable the Redundancy?

Which case was it you are referring to?  In this specific case with the target not starting, redundancy would never be enabled because we haven't gotten far enough.

Also, what I have now is that a BMC can still be active with redundancy enabled if it is in the quiesce state (of course not if it's this app that dies, but in the other cases).  Then as mentioned above the HMC could either try a reboot or do the failover.

Reply:
This PR can proceed without waiting for my comments. we could discuss offline.

I am considering hypothetical scenarios based on "the BMC will be quiesced without the role property on D-Bus." For example, alongside the RBMC manager service, another service could fail to start, causing the A/P-BMC to be quiesced. In this case, the HMC would be responsible for recovering the A/P-BMC.

Without the Role property, meaning relying solely on the BMC state as in legacy systems to recover.
If recovery fails, attempt to boot the other BMC (mimicking single BMC behavior, like a failover)."

I will look at the workshop MoM, we could discussed this already.

Reply:
I removed  the throw.  I have a story for adding error logging (want to wait for the upcoming phosphor-logging events code to be done, and also see if I can get this upstream) so eventually there will be an error log.